unified_planning/io/pddl_reader.py

```diff
@@ -582,19 +640,20 @@ def _parse_subtasks(
     def _check_if_object_type_is_needed(self, domain_res) -> bool:
         for p in domain_res.get("predicates", []):
             for g in p[1]:
-                if len(g) <= 1 or g[1] == "object":
+                if len(g) <= 1 or g[1] == Object:
                     return True
         for p in domain_res.get("functions", []):
             for g in p[1]:
-                if len(g) <= 1 or g[1] == "object":
+                if len(g) <= 1 or g[1] == Object:
                     return True
         for g in domain_res.get("constants", []):
-            if len(g) <= 1 or g[1] == "object":
+            if len(g) <= 1 or g[1] == Object:
                 return True
         for a in domain_res.get("actions", []):
             for g in a.get("params", []):
-                if len(g) <= 1 or g[1] == "object":
+                if len(g) <= 1 or g[1] == Object:
                     return True
+        # TODO probably missing spots there + is this something we really want
```

Comment:
This is something that should be handled as well. Either by :
 - completing this (to include methods, tasks ...). This would not fix the issue of the incorrect special treatment of "object" though.
 - having a special top type that is the root of the type hierarchy (often called the "Any"/"Top" type in programming languages). This could substantially simplify parsing by having the `parameter` parser always provide a type (no explicit type meaning the top type which would remove the need for the statements like `g[1] if len(g) > 1 else "object"`).

Option 2 should be the object of a distinct PR.

Reply:
I propose to go with option 1 in this PR (if it is something easy to do) and, as you said, go with option 2 in a new PR.

Reply:
Commited option 1. I can draft a PR implementing option 2 once this one is merged.